### ISSUE TITLE
State machine pattern catalogue

### ISSUE THREAD

**Issue by StoneCypher** [StoneCypher/fsl#1128]
Actually a lot of design patterns can be represented in state machines, and it's useful to see how.  Spell it out.

Blocked by #1052

**Comment by StoneCypher** [comment]
Superseded — folded into the FSL manual / taxonomy / curriculum program. Closing per the approved triage ledger (`notes/superpowers/eras/dispositions/fsl-triage.md`).
